modules/solid_mechanics/src/timeintegrators/CentralDifferenceDirect.C

```diff
@@ -0,0 +1,194 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// MOOSE includes
+#include "Assembly.h"
+#include "CentralDifference.h"
+#include "CentralDifferenceDirect.h"
+#include "MooseTypes.h"
+#include "NonlinearSystem.h"
+#include "FEProblem.h"
+// libMesh includes
+#include "libmesh/nonlinear_solver.h"
+#include "libmesh/sparse_matrix.h"
+#include "DirichletBCBase.h"
+
+registerMooseObject("MooseApp", CentralDifferenceDirect);
+
+InputParameters
+CentralDifferenceDirect::validParams()
+{
+  InputParameters params = ExplicitTimeIntegrator::validParams();
+
+  params.addClassDescription(
+      "Implementation of Explicit/Forward Euler without invoking any of the nonlinear solver");
+
+  params.addParam<bool>("use_constant_mass",
+                        false,
+                        "If set to true, will only compute the mass matrix in the first time step, "
+                        "and keep using it throughout the simulation.");
+  params.addParam<TagName>("mass_matrix_tag", "mass", "The tag for the mass matrix");
+
+  return params;
+}
+
+CentralDifferenceDirect::CentralDifferenceDirect(const InputParameters & parameters)
+  : ExplicitTimeIntegrator(parameters),
+    _constant_mass(getParam<bool>("use_constant_mass")),
+    _mass_matrix(getParam<TagName>("mass_matrix_tag"))
+{
+  _fe_problem.setUDotOldRequested(true);
+  _fe_problem.setUDotDotRequested(true);
+  _fe_problem.setUDotDotOldRequested(true);
+}
+
+void
+CentralDifferenceDirect::computeTimeDerivatives()
+{
+  /*
+  Because this is called in NonLinearSystemBase
+  this should not actually compute the time derivatives.
+  Calculating time derivatives here will cause issues for the
+  solution update.
+  */
+  return;
+}
+
+void
+CentralDifferenceDirect::solve()
+{
+
+  // Getting the tagID for the mass matrix
+  auto mass_tag = _sys.subproblem().getMatrixTagID(_mass_matrix);
+
+  // Reset iteration counts
+  _n_nonlinear_iterations = 0;
+  _n_linear_iterations = 0;
+
+  _current_time = _fe_problem.time();
+
+  // Set time to the time at which to evaluate the residual
+  _fe_problem.time() = _fe_problem.timeOld();
+  _nonlinear_implicit_system->update();
+
+  // Compute the residual
+  _explicit_residual.zero();
+  _fe_problem.computeResidual(
+      *_nonlinear_implicit_system->current_local_solution, _explicit_residual, _nl.number());
```

Comment:
could you help me understand why this _explicit_residual is Fext-Fint in the equation?

Reply:
It will make more sense once I get some test input files pushed in. I've stalled on this PR while I am at WCCM. 
This is the case because Fint will be the residual returned from your stress divergence only. Fext will return any relevant nodal BCs

Reply:
There will be no inertia kernel in the input file

Reply:
> Fint will be the residual returned from your stress divergence only. Fext will return any relevant nodal BCs

That makes sense.

> There will be no inertia kernel in the input file

How do you tag the mass matrix then?

Reply:
I've added a test to the PR. This shows how the mass matrix is tagged